dns_server/src/config.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::Parser;
+use directories::UserDirs;
+
+// TODO: Allow using with testnet
+
+#[derive(Parser, Debug)]
+pub struct DnsServerConfig {
+    /// The path to the data directory
+    #[clap(long, default_value_t = default_data_dir())]
+    pub datadir: String,
+
+    /// IP address and UDP port to listen on
+    #[clap(long, default_values_t = vec!["[::]:53".to_string()])]
+    pub bind_addr: Vec<String>,
+
+    /// Optional list of initial node addresses to connect
+    #[clap(long)]
+    pub add_node: Vec<String>,
+
+    /// Hostname of the DNS seed
+    #[clap(long)]
+    pub host: String,
+
+    /// Hostname of the nameserver
+    #[clap(long)]
+    pub ns: String,
+
+    /// E-Mail address reported in SOA records (RNAME)
+    #[clap(long)]
+    pub mbox: String,
+}
+
+const DEFAULT_DATA_DIR_NAME: &str = ".mintlayer";
```

Comment:
Is it normal to use the same data directory for the node and the dns server?

Reply:
I switched to using `BaseDirs::data_dir` instead (on Linux this is `$HOME/.local/share/mintlayer_dns_server`).

Reply:
I would rather keep it in $HOME... I hate it when things go into .local/share. That's a personal preference though... is there a reason for going in there?

Reply:
There is a specification for this: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
Some random article from the Internet: https://0x46.net/thoughts/2019/02/01/dotfile-madness/
And it's mentioned in our issues: https://github.com/mintlayer/mintlayer-core/issues/498

Maybe we should just remove default from the `datadir` argument (make it mandatory)?

Reply:
I never realized people hated this... I actually like it. But sure, let's follow that standard.

But no, we can't just force people to use that directory without the option to put it somewhere else. There can be all kinds of issues, like diskspace or permission issues.